README.md

```diff
@@ -118,3 +101,175 @@ Remote debugging using :2000
+```
+arm-none-eabi/bin/ld: region `rom' overflowed by 4088 bytes
+Memory region         Used Size  Region Size  %age Used
+             rom:      135160 B       128 KB    103.12%
+             ram:        5708 B        20 KB     27.87%
+collect2: error: ld returned 1 exit status
+```
+
+This is not unexpected, as some hardware probe have limited space for firmware. You can get around it by disabling some features or targets support:
+```sh
+meson configure build -Dtargets=cortexm,stm -Drtt_support=false
+```
+
+#### Building Black Magic Debug App
+
+The Black Magic Debug App (BMDA) is always built by default, even for firmware builds. So long as all its dependencies are found, you can find the executable under the `build` directory, named simply `blackmagic`.
+
+If you wish to build only BMDA, you can set the hardware `probe` option to an empty string `-Dprobe=''`, this is the default value:
+```sh
+meson setup build
+```
+
+You now should have a `build` directory from where you can build the app, this is also where your executable will appear.
+
+The command `meson compile` will build the default targets, you may omit `-C build` if you run the command from within the `build` directory:
+```sh
+meson compile -C build
+```
+
+You should now see the resulting executable in `build`:
+ - `blackmagic`
+
+### Changing the build configuration
+
+You may change the configuration at any time after configuring the project, if may also change the default options while configuring for the first time.
+
+Changing options at configure time:
+```sh
+meson setup build -Ddebug_output=true
+```
+
+Changing options after configuration:
+```sh
+meson configure build -Dtargets=cortexm,stm
+```
+Alternatively, you can use the equivalent but more verbose `meson setup` command:
+```sh
+meson setup build --reconfigure -Dtargets=cortexm,stm
+```
+
+Keep in mind that when changing the configured probe, the other default options will not change, and you may end up with a configuration that does not make sense for it. In such cases it's best to use the *cross-file* for the probe, not just change the `probe` option. For this you will need to use the `meson setup` command again:
+```sh
+meson setup build --reconfigure --cross-file=cross-file/bluepill.ini
+```
+
+When changing options after configuration, you may omit the argument `build` if you are running the command from within the `build` directory.
+
+You can have multiple build directories! So if you are regularly building firmware for multiple probes we would recommend keeping an individual build directory configured for each one.
+
+### Working with an existing clone (used before the new meson build system was introduced)
+
+If you are working with an existing clone of the project where you used the old `make` build system, or simply initialized the submodules, you may encounter issues while trying the new `meson` workflow, this may look something like:
```

Comment:
```suggestion
If you are working with an existing clone of the project where you used the old Makefile build system, or simply initialized the submodules, you may encounter issues while trying the new Meson workflow, this may look something like:
```